lib/fluent/plugin/out_sumologic.rb

```diff
@@ -195,36 +195,20 @@ def sumo_key(sumo_metadata, chunk)
     source_host = sumo_metadata['host'] || @source_host
     source_host = extract_placeholders(source_host, chunk) unless source_host.nil?
 
-    "#{source_name}:#{source_category}:#{source_host}"
+    fields = sumo_metadata['fields'] || ""
+    fields = extract_placeholders(fields, chunk) unless fields.nil?
+
+    "#{source_name}:#{source_category}:#{source_host}:#{fields}"
   end
 
   # Convert timestamp to 13 digit epoch if necessary
   def sumo_timestamp(time)
     time.to_s.length == 13 ? time : time * 1000
   end
 
-  def sumo_fields(sumo_metadata)
-    fields = sumo_metadata['fields'] || ""
-    Hash[
-        fields.split(',').map do |pair|
-          k, v = pair.split('=', 2)
-          [k, v]
-        end
-    ]
-  end
-
-  def dump_collected_fields(log_fields)
```

Comment:
this function will likely come back after we do https://github.com/SumoLogic/fluentd-output-sumologic/pull/53